pkg/cosign/kubernetes/apis/v1alpha1/clusterimagepolicy_types.go

```diff
@@ -0,0 +1,70 @@
+// +kubebuilder:validation:Optional
+package v1alpha1
+
+import (
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"knative.dev/pkg/apis"
+)
+
+// +kubebuilder:object:root=true
+// +kubebuilder:resource:path=clusterimagepolicies,scope=Cluster
+// +kubebuilder:storageversion
+type ClusterImagePolicy struct {
+	metav1.TypeMeta   `json:",inline"`
+	metav1.ObjectMeta `json:"metadata,omitempty"`
+	Spec              ClusterImagePolicySpec `json:"spec,omitempty"`
+}
+
+var (
+	_ apis.Validatable = (*ClusterImagePolicy)(nil)
+	_ apis.Defaultable = (*ClusterImagePolicy)(nil)
+)
+
+type ClusterImagePolicySpec struct {
+	Images []ImagePattern `json:"images,omitempty"`
+}
+
+type ImagePattern struct {
+	Pattern     string      `json:"pattern,omitempty"`
+	Authorities []Authority `json:"authorities,anyOf,omitempty"`
+}
+
+type Authority struct {
+	Key     KeyRef     `json:"key,omitempty"`
+	Keyless KeylessRef `json:"keyless,omitempty"`
+	Sources []Source   `json:"source,anyOf,omitempty"`
+	CTLog   TLog       `json:"ctlog,omitempty"`
+}
+
+type KeyRef struct {
+	SecretRef SecretRef `json:"secretRef,omitempty"`
+	Data      string    `json:"data,omitempty"`
+	KMS       string    `json:"kms,omitempty"`
+}
+
+type SecretRef struct {
+	Name string `json:"name,omitempty"`
```

Comment:
I suggest you use instead *v1.SecretReference from ``k8s.io/api/core/v1` instead of creating a new type.